### PR TITLE
chore(deps): add wanted dependencies to remove yarn warnings

### PR DESCRIPTION
**Pull request by kelset**
## Summary

I wanted to start working on a thing but this barrage of warnings was very annoying so ended up doing this instead: a very small PR to take care of some warnings during yarn install.

It doesn't change anything (the versions are the ones already used all around the repo), just makes yarn happier.

Also, while doing that I found a dependency that was lying there unused for 2 years so took care of it.

## Changelog

[General] [Fixed] - add wanted dependencies to remove yarn warnings

### Before

Screenshot 2022-10-26 at 10 53 32

### After

Screenshot 2022-10-26 at 10 52 19

### CODE REVIEW

File: packages/hermes-inspector-msggen/package.json
Comment (robhogan) [inline packages/hermes-inspector-msggen/package.json:21]:
```suggestion
    "@babel/cli": "^7.14.0",
    "@babel/core": "^7.14.0",
```

File: packages/react-native-bots/package.json
Comment (robhogan) [inline packages/react-native-bots/package.json:8]:
```suggestion
    "@seadub/danger-plugin-eslint": "^3.0.2",
    "danger": "^11.0.2",
    "eslint": "^8.19.0",
```

**Review by robhogan — APPROVED** [review]
We have a convention (internally enforced, possibly?) of alphabetically ordering dependencies but otherwise LGTM - thanks @kelset

**Comment by kelset** [comment]
@robhogan no problem, thanks for the suggestions, I've applied them! Is there some tooling that we could use to automate this? Kind of like prettier but for package.json? 😅

**Comment by robhogan** [comment]
> @robhogan no problem, thanks for the suggestions, I've applied them! Is there some tooling that we could use to automate this? Kind of like prettier but for package.json? 😅

Yeah we should, tbh I thought we had an autofix lint rule for it already, but it looks like we don't. Yarn and NPM auto-sort whenever they modify the file iirc, but it looks like we don't have any check - [`package-json/sort-collections`](https://github.com/zetlen/eslint-plugin-package-json/blob/master/docs/rules/sort-collections.md) would work. CC @cipolleschi

**Comment by robhogan** [comment]
Reopening as this was reverted due to an (apparently) missing internal dependency that previously worked transitively.

Meta folks: See D40716713 before re-landing.

**Comment by kelset** [comment]
@robhogan quick check, anything I can do to help get this in? do you want me to rebase it on top of main?

**Comment by robhogan** [comment]
> @robhogan quick check, anything I can do to help get this in? do you want me to rebase it on top of main?

Sorry about this, nothing you can do from the OSS end. Removing the `@reactions/component` dependency is tripping up Flow internally for reasons that are unclear at the moment, some config/resolution bug - we're working on it. 

*If* it'd help you to get this in, you could submit a new PR (has to be new - this one can't be re-imported now it's been committed) without the `@reactions/component` removal, that should be a straightforward merge.

Otherwise, I'm hoping we can unblock this within a few days.

**Comment by kelset** [comment]
@robhogan here you go: https://github.com/facebook/react-native/pull/35122

**Comment by kelset** [comment]
the other one got merged, so I'll close this one off

**Pull request closed without merging**
